draft-irtf-cfrg-opaque.md

```diff
@@ -1397,28 +1398,27 @@ Steps:
 4. Output concat(dh1, dh2, dh3)
 ~~~
 
-Using this shared secret, further keys used for encryption and authentication are
-computed using the following function.
-
 ~~~
 DeriveKeys(ikm, preamble)
 
 Input:
 - ikm, input key material.
-- preamble, the transcript as defined by Preamble().
+- preamble, the protocol transcript with identities and messages.
 
 Output:
-- Km2, a MAC authentication key.
-- Km3, a MAC authentication key.
 - session_key, the shared session secret.
+- server_mac, the server's MAC.
+- client_mac, the client's MAC.
 
 Steps:
 1. prk = Extract("", ikm)
 2. handshake_secret = Derive-Secret(prk, "HandshakeSecret", Hash(preamble))
 3. session_key = Derive-Secret(prk, "SessionKey", Hash(preamble))
 4. Km2 = Derive-Secret(handshake_secret, "ServerMAC", "")
 5. Km3 = Derive-Secret(handshake_secret, "ClientMAC", "")
-6. Output (Km2, Km3, session_key)
+6. server_mac = MAC(Km2, Hash(preamble))
```

Comment:
Why this change? (At some point we need to stop superfluous changes.)

Reply:
The Km keys are only needed for that purpose, so we might as well use them here directly. It avoids having these lines in `ClientFinalize()` and `Response()`, which saves a couple of lines there.

Reply:
Well, the function name is called `DeriveKeys`, but now it's doing more than that. Should we name the function if we're going to keep this change?